includes/class-shortcodes.php

```diff
@@ -448,17 +423,65 @@ public function sc_listings_featured( $atts ) {
             'publish',
             WPBDP_POST_TYPE
         );
-        $featured = $wpdb->get_col( $q );
-
+		$featured   = $wpdb->get_col( $q );
+		$query_args = array(
+			'post__in' => ! empty( $featured ) ? $featured : array( 0 ),
+			'orderby'  => 'post__in',
+		);
+		$query_args = $this->process_category_atts( $sc_atts, $query_args );
         return $this->display_listings(
-            array(
-                'post__in'  => ! empty( $featured ) ? $featured : array( 0 ),
-                'orderby'   => 'post__in',
-            ),
-            $atts
+			$query_args,
+            $sc_atts
         );
     }
 
+	/**
+	 * Process category query attributes.
+	 *
+	 * @param array $atts Shortcode attributes.
+	 * @param array $query_args The query args used to search based on attributes.
+	 *
+	 * @since x.x
+	 *
+	 * @return array $query_args
+	 */
+	private function process_category_atts( $atts, $query_args ) {
```

Comment:
How about we return $query_args by reference? That simplifies it a bit to not have to return the array.